xcm/xcm-simulator/fuzzer/src/fuzz.rs

```diff
@@ -104,46 +144,71 @@ pub fn relay_ext() -> sp_io::TestExternalities {
 pub type RelayChainPalletXcm = pallet_xcm::Pallet<relay_chain::Runtime>;
 pub type ParachainPalletXcm = pallet_xcm::Pallet<parachain::Runtime>;
 
-fn run_one_input(mut data: &[u8]) {
+fn run_input(xcm_messages: [XcmMessage; 5]) {
 	MockNet::reset();
-	if let Ok(m) = Xcm::decode_all_with_depth_limit(MAX_XCM_DECODE_DEPTH, &mut data) {
-		#[cfg(not(fuzzing))]
-		{
-			println!("Executing message {:?}", m);
+
+	#[cfg(not(fuzzing))]
+	println!();
+
+	for xcm_message in xcm_messages {
+		if xcm_message.source % 4 == 0 {
+			// We get the destination for the message
+			let parachain_id = (xcm_message.destination % 3) + 1;
+			let destination: MultiLocation = Parachain(parachain_id).into();
+			#[cfg(not(fuzzing))]
+			{
+				println!("  source:      Relay Chain");
+				println!("  destination: Parachain {parachain_id}");
+				println!("  message:     {:?}", xcm_message.message);
+			}
+			Relay::execute_with(|| {
+				assert_ok!(RelayChainPalletXcm::send_xcm(Here, destination, xcm_message.message));
+			})
+		} else {
+			// We get the source's execution method
+			let execute_with = match xcm_message.source % 4 {
+				1 => ParaA::execute_with,
+				2 => ParaB::execute_with,
+				_ => ParaC::execute_with,
+			};
+			// We get the destination for the message
+			let destination: MultiLocation = match xcm_message.destination % 4 {
+				n @ 1..=3 => (Parent, Parachain(n)).into(),
+				_ => Parent.into(),
+			};
+			#[cfg(not(fuzzing))]
+			let destination_str = match xcm_message.destination % 4 {
+				n @ 1..=3 => format!("Parachain {n}"),
+				_ => "Relay Chain".to_string(),
+			};
+			#[cfg(not(fuzzing))]
+			{
+				println!("  source:      Parachain {}", xcm_message.source % 4);
+				println!("  destination: {}", destination_str);
+				println!("  message:     {:?}", xcm_message.message);
+			}
```

Comment:
nit: Can clean this up a little:

```suggestion
			#[cfg(not(fuzzing))]
			{
			    let destination_str = match xcm_message.destination % 4 {
	 			    n @ 1..=3 => format!("Parachain {n}"),
				    _ => "Relay Chain".to_string(),
			    };
				println!("  source:      Parachain {}", xcm_message.source % 4);
				println!("  destination: {}", destination_str);
				println!("  message:     {:?}", xcm_message.message);
			}
```